drop_service/views.py

```diff
@@ -0,0 +1,109 @@
+
+import json
+import uuid
+from email.utils import formatdate
+from functools import partial
+from time import time, mktime
+
+import dateparser
+
+from django.conf import settings
+from django.http import HttpResponse, HttpResponseNotModified
+from django.views.generic import View
+from django.views.decorators.csrf import csrf_exempt
+from django.utils.decorators import method_decorator
+
+from rest_framework import status
+
+from . import monitoring
+from .models import Drop
+from .util import check_drop_id, set_last_modified
+
+
+def log_request(start_time, method, status_code):
+    time_since = time() - start_time
+    monitoring.REQUEST_TIME.labels({'method': method, 'status': status_code}).observe(time_since)
+
+
+def error(msg, status=status.HTTP_400_BAD_REQUEST):
+    return HttpResponse(json.dumps({'error': msg}), status=status)
+
+
+class DropView(View):
+    http_method_names = ['get', 'head', 'post']
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super().dispatch(request, *args, **kwargs)
+
+    def get(self, request, drop_id):
+        log = partial(log_request, time(), request.method)
+        if not check_drop_id(drop_id):
+            log(status.HTTP_400_BAD_REQUEST)
+            return error('Invalid drop id')
+
+        drops = Drop.objects.filter(drop_id=drop_id)
+        if not drops:
+            log(status.HTTP_204_NO_CONTENT)
+            return HttpResponse(status=status.HTTP_204_NO_CONTENT)
+
+        have_since, since = self.get_if_modified_since()
+        if have_since:
+            drops = drops.filter(created_at__gt=since)
+            if not drops:
+                log(status.HTTP_304_NOT_MODIFIED)
+                return HttpResponseNotModified()
+
+        if request.method == 'GET':
```

Comment:
Yes, HEAD requests are routed to the get function by default.

> By default, a HEAD request will be delegated to get(). If you need to handle HEAD requests in a different way than GET, you can override the head() method.
> https://docs.djangoproject.com/ja/1.9/ref/class-based-views/base/#django.views.generic.base.View.dispatch

I think I'll just split this into get() and head(), that'd be cleaner.